packages/trpc/server/routers/viewer/slots.ts

```diff
@@ -89,6 +99,62 @@ const checkIfIsAvailable = ({
   const slotEndTime = time.add(eventLength, "minutes").utc();
   const slotStartTime = time.utc();
 
+  const fixedDateOverrides: {
+    start: Date;
+    end: Date;
+  }[] = [];
+
+  //check if date override for slot exists
+  let dateOverrideExist = false;
+
+  if (
+    dateOverrides.find((date) => {
+      const utcOffset = organizerTimeZone ? dayjs.tz(date.start, organizerTimeZone).utcOffset() * -1 : 0;
+
+      if (
+        dayjs(date.start).add(utcOffset, "minutes").format("YYYY MM DD") ===
+        slotStartTime.format("YYYY MM DD")
+      ) {
+        dateOverrideExist = true;
+        if (dayjs(date.start).add(utcOffset, "minutes") === dayjs(date.end).add(utcOffset, "minutes")) {
+          return true;
+        }
+        if (
+          slotEndTime.isBefore(dayjs(date.start).add(utcOffset, "minutes")) ||
+          slotEndTime.isSame(dayjs(date.start).add(utcOffset, "minutes"))
+        ) {
+          return true;
+        }
+        if (slotStartTime.isAfter(dayjs(date.end).add(utcOffset, "minutes"))) {
+          return true;
+        }
+      }
+    })
+  ) {
+    // slot is not within the date override
+    return false;
+  }
```

Comment:
NIT: Personally not a fan of putting big functions in an IF like this would rather this was extracted 

```suggestion
  const foundDateOverride = dateOverrides.find((date) => {
    const utcOffset = organizerTimeZone ? dayjs.tz(date.start, organizerTimeZone).utcOffset() * -1 : 0;

    if (
      dayjs(date.start).add(utcOffset, "minutes").format("YYYY MM DD") === slotStartTime.format("YYYY MM DD")
    ) {
      dateOverrideExist = true;
      if (dayjs(date.start).add(utcOffset, "minutes") === dayjs(date.end).add(utcOffset, "minutes")) {
        return true;
      }
      if (
        slotEndTime.isBefore(dayjs(date.start).add(utcOffset, "minutes")) ||
        slotEndTime.isSame(dayjs(date.start).add(utcOffset, "minutes"))
      ) {
        return true;
      }
      if (slotStartTime.isAfter(dayjs(date.end).add(utcOffset, "minutes"))) {
        return true;
      }
    }
  });

  if (foundDateOverride) {
    // slot is not within the date override
    return false;
  }

```